tasks/backfill_gh_app_installations.py

```diff
@@ -0,0 +1,144 @@
+import logging
+
+import shared.torngit as torngit
+from sqlalchemy import String
+from sqlalchemy.orm.session import Session
+
+from app import celery_app
+from celery_config import backfill_gh_app_installations
+from database.models.core import GithubAppInstallation, Owner, Repository
+from services.owner import get_owner_provider_service
+from tasks.base import BaseCodecovTask
+
+log = logging.getLogger(__name__)
+
+
+class BackfillGHAppInstallationsTask(
+    BaseCodecovTask, name=backfill_gh_app_installations
+):
+    # Looping and adding all repositories in the installation app
+    async def add_repos_service_ids_from_provider(
+        self,
+        db_session: Session,
+        ownerid: int,
+        owner_service: torngit.base.TorngitBaseAdapter,
+        gh_app_installation: GithubAppInstallation,
+    ):
+        repos = await owner_service.list_repos_using_installation()
+        if repos:
+            # Fetching all repos service ids we have for that owner in the DB
+            repo_service_ids_in_db = [
+                repo.service_id
+                for repo in db_session.query(Repository.service_id)
+                .filter_by(ownerid=ownerid)
+                .all()
+            ]
+
+            # Add service ids from provider that we have DB records for to a list
+            new_repo_service_ids = set()
+            current_repo_service_ids = set(
+                gh_app_installation.repository_service_ids or []
+            )
+            for repo in repos:
+                repo_data = repo["repo"]
+                service_id = repo_data["service_id"]
+                if service_id and service_id in repo_service_ids_in_db:
+                    new_repo_service_ids.add(service_id)
+            log.info(
+                "Added the following repo service ids to this gh app installation",
+                extra=dict(
+                    ownerid=ownerid,
+                    installation_id=gh_app_installation.installation_id,
+                    new_repo_service_ids=list(new_repo_service_ids),
+                ),
+            )
+            gh_app_installation.repository_service_ids = list(
+                current_repo_service_ids.union(new_repo_service_ids)
+            )
+            db_session.commit()
+
+    async def run_async(
+        self, db_session: Session, ownerid: String, service: String, *args, **kwargs
+    ):
+        # Check if service isn't github
+        if service != "github":
+            log.info(
+                "No installation work needed for non-github orgs",
+                extra=dict(ownerid=ownerid),
+            )
+            return {"successful": True, "reason": "no installation needed"}
+
+        # Check if the owner exists
+        owner: Owner = (
+            db_session.query(Owner).filter_by(ownerid=ownerid, service=service).first()
+        )
+
+        if not owner:
+            log.exception(
```

Comment:
I think we should use `log.error` instead of `log.exception` it seems `exception` is reserved for use from within an `except` clause.

Reply:
Gotchaa, I got some feedback before from Gio to prefer error from exception in this PR, https://github.com/codecov/worker/pull/256#discussion_r1481451117. I'm still unclear when we should use each or the other, do you have a rule of thumb to know which one to use?